### PR TITLE
Extend EIR WWHP to Air-source

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
The EIR-formulated water-to-water heat pump model was added in EnergyPlus 9.1.0 `HeatPump:WaterToWater:EIR:Cooling` and `...Heating`. This pull request adds an air-source option to model an air-to-water central plant heat pump. The existing objects will be renamed to `HeatPump:PlantLoop:EIR:Cooling` (and Heating).

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: design/FY2019/AirSource_EIR_WaterHeatPump.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+Since there is an already-existing EIR-WWHP model, this work should integrated with it rather than creating a new object. The current object is:
+
+- HeatPump:WaterToWater:EIR:(Cooling/Heating)
+
+This model is a water-source heat pump, however, the objective of this adapt the model to be able to operate as an air-source heat pump. The EIR formulation will remain the same as the current model, as shown in the equation.
+
+```CapacityMultiplier(temperature) = a + b*(T_load) + c*(T_load^2) + d*(T_source) + e*(T_source^2) + f*(T_load*T_source)```
+
+This current project will rename this existing object to allow for water source and air source operation. It will also adapt the input structure of the exisiting object so that no new objects are created.
```
Comment (Myoldmopar) [inline design/FY2019/AirSource_EIR_WaterHeatPump.md:28]: *existing

File: design/FY2019/AirSource_EIR_WaterHeatPump.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+- Rename the existing EIR-WWHP from:
+
+  ```HeatPump:WaterToWater:EIR:(Cooling/Heating)```
+
+  to:
+
+  ```HeatPump:PlantLoop:EIR:(Cooling/Heating)```
+
+  "PlantLoop" could also be replaced by suggestions from the reviewers. Perhaps, "WaterSide" or "WaterSideLoad"?
```
Comment (Myoldmopar) [inline design/FY2019/AirSource_EIR_WaterHeatPump.md:40]: It is really hard to come up with a better name.  Unless we just want to take out the middle descriptor, leaving out plant loop entirely.  But then it doesn't really describe where it should be used.  In the very long run, if a single heat pump could be used for providing heating and cooling to both air and water, then we could just have HeatPump:EIR...but for now this seems fine.

File: doc/engineering-reference/src/simulation-models-encyclopedic-reference-002/air-system-compound-component-groups.tex [hunk 3]
```diff
@@ -2554,9 +2554,9 @@ \subsubsection{References}\label{references-1}
...
+This section describes the EIR formulated model for plant loop heat pumps, which can be configured as either a water-source or air-source object. (Object names: HeatPump:PlantLoop:EIR:Cooling \& HeatPump:PlantLoop:EIR:Heating). In general, these heat pump objects are treated the same as the other heat pump models by the plant loop.  Even though a heat pump is generally a single load coil and a single source coil with a reversing valve, in EnergyPlus, the paradigm is to split the operation into two separate coils, a heating and a cooling.  It is certainly possible to connect the load side of both of these to a single plant loop if the controls are established properly.  For water-source operation the source side of the coils are often placed on a single loop with some form of condensing supply, either a ground loop heat exchanger or a cooling tower (or a combination of the two). For air-source operation, the outdoor air node is used on the source-side inlet node.
```
Comment (Myoldmopar) [inline doc/engineering-reference/src/simulation-models-encyclopedic-reference-002/air-system-compound-component-groups.tex:2559]: :+1:

File: doc/input-output-reference/src/overview/group-plant-equipment.tex [hunk 4]
```diff
@@ -5048,75 +5048,80 @@ \subsubsection{Outputs}\label{outputs-13-003}
...
 
-\paragraph{Field: Load Side Outlet Node Name}\label{wwhpeir_inputs_load_outlet_node}
+\paragraph{Field: Load Side Outlet Node Name}\label{plhp_eir_inputs_load_outlet_node}
 
 This alpha field contains the heat pump's load side outlet node name.
 
-\paragraph{Field: Source Side Inlet Node Name}\label{wwhpeir_inputs_source_inlet_node}
+\paragraph{Field: Condenser Type}\label{plhp_eir_condenser_type}
+
+This alpha field contains the heat pump's condenser type. Acceptable inputs are ``WaterSource" or ``AirSource".
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-plant-equipment.tex:5071]: :+1:

File: doc/input-output-reference/src/overview/group-plant-equipment.tex [hunk 5]
```diff
@@ -5145,82 +5151,119 @@ \subsubsection{Inputs}\label{wwhpeir_inputs}
...
 
 This output variable represents the outlet temperature on the source side connection of this coil.
 
-\paragraph{Water to Water Heat Pump Load Side Mass Flow Rate {[}kg/s{]}}\label{eir_wwhp_outputs_load_flow}
+\paragraph{Plant Loop EIR Heat Pump Load Side Mass Flow Rate {[}kg/s{]}}\label{plhp_eir_outputs_load_flow}
 
 This output variable represents the average load side mass flow rate during the specified reporting window.
 
-\paragraph{Water to Water Heat Pump Source Side Mass Flow Rate {[}kg/s{]}}\label{eir_wwhp_outputs_src_flow}
+\paragraph{Plant Loop EIR Heat Pump Source Side Mass Flow Rate {[}kg/s{]}}\label{plhp_eir_outputs_src_flow}
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-plant-equipment.tex]: This report variable change will trigger an addition to the Report Variables transition support file.

File: idd/Energy+.idd.in [hunk 6]
```diff
@@ -70818,13 +70823,18 @@ HeatPump:WaterToWater:EIR:Heating,
    A3,  \field Load Side Outlet Node Name
         \required-field
         \type node
-   A4,  \field Source Side Inlet Node Name
+   A4,  \field Condenser Type
+        \type choice
+        \required-field
+        \key WaterSource
+        \key AirSource
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:70830]: Should we default to WaterSource just for fun?

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 7]
```diff
@@ -64,46 +65,51 @@
...
 
 namespace EnergyPlus {
-    namespace EIRWaterToWaterHeatPumps {
+    namespace EIRPlantLoopHeatPumps {
 
-        bool getInputsWWHP(true);
-        std::vector<EIRWaterToWaterHeatPump> eir_wwhp;
-        std::string static const __EQUIP__ = "EIRWaterToWaterHeatPump"; // NOLINT(cert-err58-cpp)
+        bool getInputsPLHP(true);
+        std::vector<EIRPlantLoopHeatPump> eir_plhp;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: I really shouldn't have done snake case here when I first added `eir_wwhp`.  If you have more changes to make, feel free to make it `eirPlhp` or even better just `heatPumps`.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 8]
```diff
@@ -64,46 +65,51 @@
...
+                                                                        this->sourceSideNodes.outlet,
+                                                                        this->sourceSideHeatTransfer,
+                                                                        this->sourceSideInletTemp,
+                                                                        this->sourceSideOutletTemp,
+                                                                        this->sourceSideMassFlowRate,
+                                                                        FirstHVACIteration);
+                    return;
+                }
+            } else if (this->airSource) {
+                this->setOperatingFlowRatesASHP();
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: :+1:

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 9]
```diff
@@ -234,7 +240,46 @@ namespace EnergyPlus {
...
+                    PlantUtilities::SetComponentFlowRate(this->loadSideMassFlowRate,
+                                                         this->loadSideNodes.inlet,
+                                                         this->loadSideNodes.outlet,
+                                                         this->loadSideLocation.loopNum,
+                                                         this->loadSideLocation.loopSideNum,
+                                                         this->loadSideLocation.branchNum,
+                                                         this->loadSideLocation.compNum);
+                }
+            }
+        }
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc:287]: Straightforward addition.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 10]
```diff
@@ -294,62 +339,74 @@ namespace EnergyPlus {
...
+                        thisLoadPlantLoop.FluidName,
+                        DataLoopNode::Node(this->loadSideNodes.inlet).Temp,
+                        thisLoadPlantLoop.FluidIndex,
+                        "PLHPEIR::simulate()");
+            } else if (this->airSource) {
+                CpSrc = Psychrometrics::PsyCpAirFnWTdb(
+                        DataEnvironment::OutHumRat,
+                        DataEnvironment::OutDryBulbTemp);
+            }
+            Real64 const sourceMCp = this->sourceSideMassFlowRate * CpSrc;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: Makes sense.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 11]
```diff
@@ -452,33 +513,47 @@ namespace EnergyPlus {
...
+                                                       Real64 &MinLoad,
+                                                       Real64 &OptLoad) {
             if (calledFromLocation.loopNum == this->loadSideLocation.loopNum) {
-                this->size();
+                this->sizeLoadSide();
+                if (this->waterSource) {
+                    this->sizeSrcSideWSHP();
+                } else if (this->airSource) {
+                    this->sizeSrcSideASHP();
+                }
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: I like this, and I could see how in the long run _maybe_ someone could push this further to get:

```
if waterLoad:
  this->sizeLoadWater()
else:
  this->sizeLoadAir()
```

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 12]
```diff
@@ -982,105 +1139,126 @@ namespace EnergyPlus {
...
+                        if (condenserType == "WATERSOURCE") {
+                            thisPLHP.waterSource = true;
+                            condenserNodeType = DataLoopNode::NodeType_Water;
+                            condenserNodeConnectionType_Inlet = DataLoopNode::NodeConnectionType_Inlet;
+                            condenserNodeConnectionType_Outlet = DataLoopNode::NodeConnectionType_Outlet;
+                        } else if (condenserType == "AIRSOURCE") {
+                            thisPLHP.airSource = true;
+                            condenserNodeType = DataLoopNode::NodeType_Air;
+                            condenserNodeConnectionType_Inlet = DataLoopNode::NodeConnectionType_OutsideAir;
+                            condenserNodeConnectionType_Outlet = DataLoopNode::NodeConnectionType_OutsideAir;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: YES this way you can still have single calls to GetOnlySingleNode.  This is the stuff.

File: tst/EnergyPlus/unit/CMakeLists.txt [hunk 13]
```diff
@@ -183,7 +183,7 @@ set( test_src
   WaterThermalTanks.unit.cc
   WaterToAirHeatPump.unit.cc
   WaterToAirHeatPumpSimple.unit.cc
-  WaterToWaterHeatPumpEIR.unit.cc
+        PlantLoopHeatPumpEIR.unit.cc
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/CMakeLists.txt]: Indentation snuck in here, I know CLion likes to indent the CMakeLists files automatically...

**Review by Myoldmopar — APPROVED** [review]
I haven't run anything or even built locally, but this is looking really great from a code review standpoint.  Feel free to comment back on anything I wrote for further discussion.

**Comment by mitchute** [comment]
@Myoldmopar thanks for the review. I just merged in develop. We'll go from there once the results come back.

**Comment by mitchute** [comment]
@Myoldmopar this is ready to go.

File: src/Transition/SupportFiles/Report Variables 9-2-0 to 9-3-0.csv [hunk 14]
```diff
@@ -0,0 +1,15 @@
+9.1.0,9.2.0,Transition notes
+0,0,These numbers should be the number of report variables in the following list (including deletes). Two columns/numbers.
```
Comment (Myoldmopar) [inline src/Transition/SupportFiles/Report Variables 9-2-0 to 9-3-0.csv]: The version numbers here don't match in line 1 and the number of variables needs to be specified in line 2.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 15]
```diff
@@ -64,46 +65,51 @@
...
+        bool getInputsPLHP(true);
+        std::vector<EIRPlantLoopHeatPump> heatPumps;
+        std::string static const __EQUIP__ = "EIRPlantLoopHeatPump "; // NOLINT(cert-err58-cpp)
 
-        void EIRWaterToWaterHeatPump::clear_state() {
-            getInputsWWHP = true;
-            eir_wwhp.clear();
+        void EIRPlantLoopHeatPump::clear_state() {
+            getInputsPLHP = true;
+            heatPumps.clear();
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: Whoever put in the `eir_wwhp` variable was a real goof.  Snake case in c++, yikes.  This is **much** better :smile:

File: idd/Energy+.idd.in [hunk 16]
```diff
@@ -70727,13 +70727,19 @@ HeatPump:WaterToWater:EIR:Cooling,
    A3,  \field Load Side Outlet Node Name
         \required-field
         \type node
-   A4,  \field Source Side Inlet Node Name
+   A4,  \field Condenser Type
+        \required-field
+        \type choice
+        \key WaterSource
+        \key AirSource
+        \default WaterSource
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:70735]: New field, defaulting back to the water source, good.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 17]
```diff
@@ -930,50 +1085,53 @@ namespace EnergyPlus {
...
+                        thisPLHP.name = UtilityRoutines::MakeUPPERCase(thisObjectName);
                         std::string loadSideInletNodeName = UtilityRoutines::MakeUPPERCase(
                                 fields.at("load_side_inlet_node_name")
                         );
                         std::string loadSideOutletNodeName = UtilityRoutines::MakeUPPERCase(
                                 fields.at("load_side_outlet_node_name")
                         );
+                        std::string condenserType = UtilityRoutines::MakeUPPERCase(
+                                fields.at("condenser_type")
+                        );
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc]: Getting the new required field, looks great.

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
This is really close.  I made a couple very small requests on the report variable transition file.  While you are in there, can you create a new 9.2-9.3 transition rules markdown file, and document the changes needed?  Soon I will push a branch with the new Fortran transition code and all the other good stuff that goes along with it, and I'll include this transition rule in there.

@mjwitte if you care to look through the IDD changes at least, that would be great.

File: idd/Energy+.idd.in [hunk 18]
```diff
@@ -70727,13 +70727,19 @@ HeatPump:WaterToWater:EIR:Cooling,
...
+   A5,  \field Source Side Inlet Node Name
         \required-field
         \type node
-   A5,  \field Source Side Outlet Node Name
+   A6,  \field Source Side Outlet Node Name
         \required-field
         \type node
-   A6,  \field Companion Heat Pump Name
+   A7,  \field Companion Heat Pump Name
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: \reference PLHPHeatingNames

File: idd/Energy+.idd.in [hunk 19]
```diff
@@ -70818,13 +70824,19 @@ HeatPump:WaterToWater:EIR:Heating,
...
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: \reference PLHPCoolingNames

File: src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md [hunk 20]
```diff
@@ -12,3 +12,40 @@ This will eventually become a more structured file, but currently it isn't clear
...
+Water to Water Heat Pump Electric Power
+Water to Water Heat Pump Electric Energy
+Water to Water Heat Pump Load Side Mass Flow Rate
+Water to Water Heat Pump Source Side Mass Flow Rate
+```
+
+*9.3.0*
+```
+Plant Loop Heat Pump Load Side Heat Transfer Rate
+Plant Loop Heat Pump Load Side Heat Transfer Energy
```
Comment (mjwitte) [inline src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md]: In general we try to have generic output variables names that are similar across different equipment (e.g. all chiller types have the same output variable names, except maybe a few oddballs). Other equipment, such as HeatPump:WaterToWater:EquationFit:Cooling use the old Water to Water Heat Pump * names here. But I see how that doesn't work when it's air-source. 

Maybe we want to consider changing the other equipment to match this one?  That would make the transition code easier. As it stands now, transition will change any occurrence of Water to Water Heat Pump * output variables to the new Plant Loop Heat Pump * names - it doesn't check to see what type of equipment is being referenced.

So, we either need to change these names everywhere, or we need to add smarts to transition to only change these for this type of equipment.

**Review by mjwitte — COMMENTED** [review]
In general, I'm fine with the naming here. See comments below.

File: src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md (hunk 20 above)
Comment (mitchute) [reply src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md]: Maybe we should just drop the "Plant Loop" *and* "Water to Water" on all of these outputs?

Comment (mjwitte) [reply src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md]: That's a thought. How would that align with the other heat pump objects like heat pump water heater, water-to-air, and air-to-air?

Comment (Myoldmopar) [reply src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md]: I think the answer to this will just come from what the RDD looks like in different cases.  Does unifying the variable names across different components make it easier on the user or more difficult.  The RDD doesn't give an obvious clue as to the linkage between components and output variables unless the component is actually in the variable name.  This might end up being a bigger discussion and so my recommendation is, whatever we do, not dwell too heavily on it for this particular PR.

At this moment, I'd probably support just making this "Heat Pump Load Side Foo Bar", and then maybe follow-up with a PR to unify the other heat pump objects with the same name, and evaluate the effects on the RDD over in that new PR instead of this one.

Comment (mitchute) [reply src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md]: Let's see, these following take the "Load Side" - "Source Side" approach: 

```HeatPumpWaterToWaterCOOLING.cc```
"Water to Water Heat Pump Electric Power"
"Water to Water Heat Pump Load Side Heat Transfer Rate"

```HeatPumpWaterToWaterHEATING.cc```
"Water to Water Heat Pump Electric Power"
"Water to Water Heat Pump Load Side Heat Transfer Rate"

```HeatPumpWaterToWaterSimple.cc```
"Water to Water Heat Pump Electric Power"
"Water to Water Heat Pump Load Side Heat Transfer Rate"

```PlantLoopHeatPump.cc```
"Plant Loop Heat Pump Load Side Heat Transfer Rate"
"Plant Loop Heat Pump Source Side Heat Transfer Rate

```WaterToAirHeatPump.cc```
"Heating Coil Electric Energy"
"Heating Coil Source Side Heat Transfer Energy"


This goes with specifying "Air" - "Water":

```IntegratedHeatPump.cc```
"Integrated Heat Pump Electric Power"
"Integrated Heat Pump Water Heating Rate"
"Integrated Heat Pump Air Latent Cooling Rate"


This mixes "Source Side" and "Latent/Sensible":

```WaterToAirHeatPumpSimple.cc```
"Cooling Coil Electric Energy"
"Cooling Coil Source Side Heat Transfer Energy"
"Cooling Coil Latent Cooling Energy"


This looks like it doesn't follow any convention:

```WaterThermalTanks.cc -- WaterHeater:HeatPump:PumpedCondenser```
"Water Heater On Cycle Ancillary Electric Power"
"Water Heater Heat Pump Control Tank Temperature"


Are these included?

```PackagedTerminalHeatPump.cc```
"Zone Packaged Terminal Heat Pump Electric Power"
"Zone Packaged Terminal Heat Pump Latent Cooling Rate"

```HVACMultiSpeedHeatPump.cc```
"Unitary System Electric Power"

In summary, it looks like there could be some standardization of output vars, but maybe we handle that separately.

Comment (rraustad) [reply src/Transition/OutputRulesFiles/OutputChanges9-2-0-to-9-3-0.md]: There is a difference between parent and child power/energy report variable names. It is the child that reports to the meter and the parent only echoes this data as a convenience for the user (and does not report to a meter). So when choosing report names, it is good to use the same name for say coils, Cooling Coil Total Cooling Energy, because the energy use from all coil types can be aggregated by using a single name. For parent objects this is not all that important, unless of course you want to aggregate total power/energy for systems, because the parent object is not responsible for meter reporting. For example, if Unitary System Total Cooling Energy/Rate were used for all system types (PTHP, UnitarySystem, Furnace, etc.) then a single report name could be used to report or aggregate all the information (coils and fans) in the entire simulation but would not discern between coils and fans or need to match specific meters. This is most important for UIs where they don't need to keep track of different report variable names for different system types. And for UIs I think what is important is that the magnitude of the report data matches what the meter would report. So for parent object, it would be nice to be able to use a single name to get reports, but it's not as important as for child components.

Take a report like this for example, it's more important when there are child water coils and DX coils in the same simulation to use the same name. You would not want to do this twice. For parent objects, this is not as important, but we need to think this out in case it makes sense to use the same logic for parents.

    Output:Table:Monthly,
    Coil Report,                              !- Name
    ,                                       !- Digits After Decimal
    Cooling Coil Total Cooling Energy,      !- Variable or Meter Name 1
    SumOrAverage,                           !- Aggregation Type for Variable or Meter 1
    Cooling Coil Electric Energy,           !- Variable or Meter Name 2
    SumOrAverage,                           !- Aggregation Type for Variable or Meter 2
    Cooling Coil Total Cooling Energy,      !- Variable or Meter Name 3
    HoursNonZero,                           !- Aggregation Type for Variable or Meter 3
    Cooling Coil Sensible Cooling Energy,   !- Variable or Meter Name 4
    SumOrAverage,                           !- Aggregation Type for Variable or Meter 4
    Cooling Coil Latent Cooling Energy,     !- Variable or Meter Name 5

**Comment by nrel-bot-3** [comment]
@mitchute @lgentile it has been 29 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
@mitchute something definitely broke when you pulled develop in.  Looks like the license year for one thing and then an #include that probably just needs the EnergyPlus directory inserted at the beginning.

**Comment by mitchute** [comment]
OK; I'll get it fixed up here shortly.

**Comment by Myoldmopar** [comment]
Looks like the two EIR example files failed, I wonder if a merge conflict issue caused it or something...

**Comment by mitchute** [comment]
So much for the "oh it'll be OK" handwave as I pushed it before testing those... 🤦‍♂ 

I'll look at it now.

**Comment by mitchute** [comment]
Let's see if that does it 🤞

File: idd/Energy+.idd.in [hunk 21]
```diff
@@ -70729,12 +70729,12 @@ ChillerHeater:Absorption:DoubleEffect,
        \minimum> 0.0
        \default 1.0
 
-HeatPump:WaterToWater:EIR:Cooling,
+HeatPump:PlantLoop:EIR:Cooling,
         \memo An EIR formulated water to water heat pump model, cooling operation.
-        \min-fields 15
+        \min-fields 16
```
Comment (mjwitte) [inline idd/Energy+.idd.in:70734]: @mitchute This should be 15, 'cause 15 was wrong before. But #7659 will move this to 14, so ultimately before this merges and after #7659 is in, we need to make sure this is 15. Same for the Heating object.

Comment (mitchute) [reply idd/Energy+.idd.in:70734]: @mjwitte noted. I'll make a reminder for myself to keep an eye on this.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
I can fix the IDD min-fields issue and merge if that's the only thing wrong.

**Comment by mjwitte** [comment]
@Myoldmopar Either way. If this drop in first, then #7659 won't need an IDD change.

**Comment by Myoldmopar** [comment]
Hmm this also needs the actual fortran transition rules.  @mitchute can you add those?  I pushed up the min-fields fix so once the transition code is in I'll test it and merge this.

**Comment by mitchute** [comment]
@Myoldmopar we will also have to correct it over on idd/V9-2-0....idd for this to work, right?

**Comment by mitchute** [comment]
@Myoldmopar @mjwitte I added transition rules for this here: 094c81180c67a6b12609f87ae3e3479e7d6d3b0b

It all looks correct to me, but my local testing shows that although the fields transition correctly, the field comments do not. Hopefully one of you two can point me in the right direction.

**Comment by mitchute** [comment]
@mjwitte Thanks. It works now.

**Comment by Myoldmopar** [comment]
OK, Windows box was a space issue.  I restarted it and it's running again but it may be a while before it gets back here.  I'm OK dropping this in now without waiting.  Objections?

**Comment by mitchute** [comment]
👉 ✅

**Comment by Myoldmopar** [comment]
Merging, thanks @mitchute.
